### PR TITLE
Add documentation on representations

### LINKED ISSUE #2699: Add document to describe the representation process for mentors
We should add the `exercises/shared/.docs/representations.md` file to describe to mentors how the representations are generated. This content is displayed when a mentor uses the 'Automation' tab on the website.

Information on what this file should contain can be found here: https://exercism.org/docs/building/tracks/shared-files#h-file-representations-md

### PR DESCRIPTION
**Pull request by sanderploegsma**
The `exercises/shared/.docs/representations.md` file contents are displayed in the Automations part of the website used by mentors to set up automated mentor feedback on solution sets grouped by their representations. This is explained here: https://exercism.org/docs/building/tracks/shared-files#h-file-representations-md

Fixes #2699 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by sanderploegsma** [comment]
I also noticed that the Elixir track contains more documentation on how these normalizations work: https://github.com/exercism/elixir/blob/d0f255138e0b13eb81e364e2ab6c3a9aa498e540/exercises/shared/.docs/representations.md

I'd like to add this as well, but since the representer might be overhauled soon after merging https://github.com/exercism/java-representer/pull/109 I'd like to hold off on that for now.

**Comment by sanderploegsma** [comment]
Currently this file describes the representer as it is running right now, so I'd say we merge this.

**Review by manumafe98 — APPROVED** [review]

**Pull request merged**
